### ISSUE TITLE
Backup should abort if backup.inprogress is found

### ISSUE THREAD

**Issue by fthiery** [laurent22/rsync-time-backup#79]
I tested and apparently if you launch another backup when one is already running (hence the flock advice in the doc), it is still executed. 

Why not use the fact that backup.inprogress if found to abort the backup (like a .pid file)? There's probably a good reason, but i'm interested.

Thanks

**Comment by laurent22** [comment]
This is how it works actually, if a backup.inprogress (which contains the pid) is present, the new backup operation should not start. The check is done there: https://github.com/laurent22/rsync-time-backup/blob/master/rsync_tmbackup.sh#L290

If it doesn't work for you, please could you provide more information (operating system, log file, directory structure, and any information that can help replicating the issue)

**Comment by fthiery** [comment]
I just tried

```
root@flobuntuserver:/home/fthiery# mkdir -p -- "/backup/test" ; touch "/backup/test/backup.marker"
root@flobuntuserver:/home/fthiery# rsync_tmbackup.sh /usr/lib/ /backup/test
```
Backup runs, opening another session, 
```
root@flobuntuserver:/home/fthiery# rsync_tmbackup.sh /usr/lib/ /backup/test
```
Backup runs too, even if /backup/test/backup.inprogress exists, when the first initial backup is sill in progress.

I then rebooted the system (to flush any fs cache), and ran the following in 2 shells at the nearly the same time:
```
rsync_tmbackup.sh /usr/lib/ /backup/test
```

They both run

**Comment by laurent22** [comment]
On which OS is it?

**Comment by fthiery** [comment]
Ubuntu server 16.04 LTS 64 bits

**Comment by Farstucker** [comment]
Also runing Ubuntu 16.04 LTS 64Bit, have a problem which might be related. Although i can see rsync performing the backups correctly as scheduled from cron, i check what processes are running and it seems that once an rsync process is started, it never quits once finished. Logs seem to indicate a successful backup otherwise. backup.inprogress is created with correct PID.

**Comment by luckydonald** [comment]
I get a similar feeling...
![image](https://user-images.githubusercontent.com/2737108/33225160-c84da472-d172-11e7-8da0-ec027d6f060e.png)
![image](https://user-images.githubusercontent.com/2737108/33225183-0d314620-d173-11e7-8227-0fee424f20cf.png)

**Comment by luckydonald** [comment]
@laurent22 could you replace your link with a [permalink](https://help.github.com/articles/getting-permanent-links-to-files/#press-y-to-permalink-to-a-file-in-a-specific-commit)?

**Comment by luckydonald** [comment]
```shell
luckydonald@pegasopolis:~$ screen -S back sudo /backups/rsync_tmbackup.sh /data /backups/data
[detached from 21081.back]
luckydonald@pegasopolis:~$ sudo /backups/rsync_tmbackup.sh /data /backups/data
rsync_tmbackup: /backups/data/backup.inprogress already exists - the previous backup failed or was interrupted. Backup will resume from there.
rsync_tmbackup: Previous backup found - doing incremental backup from /backups/data/2017-11-24-230002
rsync_tmbackup: Creating destination /backups/data/2017-11-25-000845
rsync_tmbackup: Starting backup...
rsync_tmbackup: From: /data/
rsync_tmbackup: To:   /backups/data/2017-11-25-000845/
rsync_tmbackup: Running command:
rsync_tmbackup: rsync -D --compress --numeric-ids --links --hard-links --one-file-system --itemize-changes --times --recursive --perms --owner --group --log-file '/root/.rsync_tmbackup/2017-11-25-000845.log' --link-dest='/backups/data/2017-11-24-230002' -- '/data/' '/backups/data/2017-11-25-000845/' | grep -E '^deleting|[^/]$'
>f..t...... somefile1
>f.st...... somefile2
>f..t...... somefile3
>f..t...... somefile4
^Crsync error: received SIGINT, SIGTERM, or SIGHUP (code 20) at rsync.c(632) [sender=3.1.1]
rsync error: received SIGINT, SIGTERM, or SIGHUP (code 20) at io.c(504) [generator=3.1.1]
rsync_tmbackup: SIGINT caught.
```
In fact, even with `sudo /backups/rsync_tmbackup.sh /data /backups/data` running in background inside a `screen`, I can still start the same command again, and  
> `Backup will resume from there.`
The background one doesn't terminate, the new doesn't terminate.

**Comment by luckydonald** [comment]
In fact, on my system `pgrep` just fails to return anything with `rsync_tmbackup`. Using parts from [this answer](https://superuser.com/a/481409) yields results:
```shell
luckydonald@pegasopolis:~$ screen -S back sudo /backups/rsync_tmbackup.sh /data /backups/data
[detached from 18573.back]
luckydonald@pegasopolis:~$ APPNAME=rsync_tmbackup
luckydonald@pegasopolis:~$ pgrep "$APPNAME"
luckydonald@pegasopolis:~$ ps -A -o pid,cmd|grep "$APPNAME" | grep -v grep |head -n 1 | awk '{print $1}'
3275
luckydonald@pegasopolis:~$ sudo cat /backups/data/backup.inprogress
18591
luckydonald@pegasopolis:~$ ps -A -o pid,cmd|grep "$APPNAME" | grep -v grep
 3275 SCREEN -S back sudo /backups/rsync_tmbackup.sh /data /backups/data
 3276 sudo /backups/rsync_tmbackup.sh /data /backups/data
 3468 bash /backups/rsync_tmbackup.sh /data /backups/data
 3651 rsync -D --compress --numeric-ids --links --hard-links --one-file-system --itemize-changes --times --recursive --perms --owner --group --log-file /root/.rsync_tmbackup/2017-11-25-001146.log --link-dest=/backups/data/2017-11-24-230002 -- /data/ /backups/data/2017-11-25-001146/
 3653 rsync -D --compress --numeric-ids --links --hard-links --one-file-system --itemize-changes --times --recursive --perms --owner --group --log-file /root/.rsync_tmbackup/2017-11-25-001146.log --link-dest=/backups/data/2017-11-24-230002 -- /data/ /backups/data/2017-11-25-001146/
 3654 rsync -D --compress --numeric-ids --links --hard-links --one-file-system --itemize-changes --times --recursive --perms --owner --group --log-file /root/.rsync_tmbackup/2017-11-25-001146.log --link-dest=/backups/data/2017-11-24-230002 -- /data/ /backups/data/2017-11-25-001146/
18573 SCREEN -S back sudo /backups/rsync_tmbackup.sh /data /backups/data
18574 sudo /backups/rsync_tmbackup.sh /data /backups/data
18591 bash /backups/rsync_tmbackup.sh /data /backups/data
18762 rsync -D --compress --numeric-ids --links --hard-links --one-file-system --itemize-changes --times --recursive --perms --owner --group --log-file /root/.rsync_tmbackup/2017-11-25-002004.log --link-dest=/backups/data/2017-11-24-230002 -- /data/ /backups/data/2017-11-25-002004/
18764 rsync -D --compress --numeric-ids --links --hard-links --one-file-system --itemize-changes --times --recursive --perms --owner --group --log-file /root/.rsync_tmbackup/2017-11-25-002004.log --link-dest=/backups/data/2017-11-24-230002 -- /data/ /backups/data/2017-11-25-002004/
18765 rsync -D --compress --numeric-ids --links --hard-links --one-file-system --itemize-changes --times --recursive --perms --owner --group --log-file /root/.rsync_tmbackup/2017-11-25-002004.log --link-dest=/backups/data/2017-11-24-230002 -- /data/ /backups/data/2017-11-25-002004/
luckydonald@pegasopolis:~$ uname -a
Linux pegasopolis 3.16.0-4-amd64 #1 SMP Debian 3.16.43-2+deb8u5 (2017-09-19) x86_64 GNU/Linux
```

**Comment by luckydonald** [comment]
Another approach would be to get the `PID` from the `backup.inprogress` file, and check if there is a process running which matches `rsync_tmbackup`:
```shell
luckydonald@pegasopolis:~$ PID=$(cat /backups/data/backup.inprogress)
luckydonald@pegasopolis:~$ echo $PID
18591
luckydonald@pegasopolis:~$  ps -o pid,cmd -p $PID | grep "$APPNAME"
18591 bash /backups/rsync_tmbackup.sh /data /backups/data
luckydonald@pegasopolis:~$
```
(Using the screen session from the post before)

Maybe checking the return status is enough?

The existing process:
```shell
luckydonald@pegasopolis:~$ ps -o pid,cmd -p $PID | grep "$APPNAME"
18591 bash /backups/rsync_tmbackup.sh /data /backups/data
luckydonald@pegasopolis:~$ echo $?
0
```
A non existing process:
```shell
luckydonald@pegasopolis:~$ ps -o pid,cmd -p 18766
  PID CMD
luckydonald@pegasopolis:~$ ps -o pid,cmd -p 99999 | grep "$APPNAME"
luckydonald@pegasopolis:~$ echo $?
1
```
A different process:
luckydonald@pegasopolis:~$ ps -o pid,cmd -p 14581
  PID CMD
14581 -bash
luckydonald@pegasopolis:~$ ps -o pid,cmd -p 14581  | grep "$APPNAME"
luckydonald@pegasopolis:~$ echo $?
1
```

**Comment by laurent22** [comment]
I think a recent pr fixed this issue. If not please comment back here

**Comment by luckydonald** [comment]
@laurent22 I can still launch another instance without anything preventing it.

Left started first, it did some cleanup first, and when doing file copying I started the second one on the right.
They both run. No stopping there.

This is was caused much instability to my server, as eventually they pile up, and the server is running out of resources. Crashing ~ every 2 days.  Disabling backup made it stable again.

![untitled](https://user-images.githubusercontent.com/2737108/41999978-95aa7aae-7a5f-11e8-99e9-49d5ecc6fa70.png)

**Comment by luckydonald** [comment]
@laurent22: 
Which PR should have fixed this?

**Comment by luckydonald** [comment]
Also, the `backup.inprogress`  file isn't deleted on abort.
Maybe add cleanup for int and term interrupts?

```sh
function cleanup {
	# remove .inprogress file
	fn_rm_file "$INPROGRESS_FILE"
}
trap cleanup INT TERM;

# also regularly at end of script:
cleanup;
``` 
[source](https://unix.stackexchange.com/a/149093/200391)

**Comment by luckydonald** [comment]
Clearly it seems that the .inprogress file was updated incorrectly.
![screen shot 2018-06-27 at 23 42 44](https://user-images.githubusercontent.com/2737108/42001292-da252f9a-7a63-11e8-8959-5aaba23e4bef.png)

**Comment by SimonHeimberg** [comment]
Pr to fix problems with lock file: #111 or #129
